src/Control/Effect/Class.hs

```diff
@@ -6,46 +6,18 @@
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE TypeOperators #-}
 
--- | Provides the 'HFunctor' and 'Effect' classes that effect types implement.
+-- | Provides the 'Effect' class that many effect types implement.
 --
 -- @since 1.0.0.0
 module Control.Effect.Class
-( HFunctor(..)
-, handleCoercible
-, Effect(..)
--- * Generic deriving of 'HFunctor' & 'Effect' instances.
-, GHFunctor(..)
+( Effect(..)
+-- * Generic deriving of 'Effect' instances.
 , GEffect(..)
 ) where
 
 import Data.Coerce
 import GHC.Generics
 
--- | Higher-order functors of kind @(* -> *) -> (* -> *)@ map functors to functors.
---
---   All effects must be 'HFunctor's.
---
--- @since 1.0.0.0
-class HFunctor h where
-  -- | Higher-order functor map of a natural transformation over higher-order positions within the effect.
-  --
-  -- A definition for 'hmap' over first-order effects can be derived automatically provided a 'Generic1' instance is available.
-  hmap :: Functor m => (forall x . m x -> n x) -> (h m a -> h n a)
-  default hmap :: (Functor m, Generic1 (h m), Generic1 (h n), GHFunctor m n (Rep1 (h m)) (Rep1 (h n))) => (forall x . m x -> n x) -> (h m a -> h n a)
-  hmap f = to1 . ghmap f . from1
-  {-# INLINE hmap #-}
-
-
--- | Thread a 'Coercible' carrier through an 'HFunctor'.
---
---   This is applicable whenever @f@ is 'Coercible' to @g@, e.g. simple @newtype@s.
---
--- @since 1.0.0.0
-handleCoercible :: (HFunctor sig, Functor f, Coercible f g) => sig f a -> sig g a
-handleCoercible = hmap coerce
```

Comment:
_All_ uses of `hmap` in the library are subsumed by the homomorphism parameter to `alg`, so we can eliminate `handleCoercible` along with the rest.